rechunker/api.py

```diff
@@ -311,6 +312,30 @@ def rechunk(
     return Rechunked(executor, plan, source, intermediate, target)
 
 
+def parse_target_chunks_from_dim_chunks(ds, target_chunks):
+    """
+    Calculate ``target_chunks`` suitable for ``rechunker.rechunk()`` using chunks defined for
+    dataset dimensions (similar to xarray's ``.rechunk()``) .
+
+    - If a dimension is missing from ``target_chunks`` then use the full length from ``ds``.
+    - If a chunk in ``target_chunks`` is larger than the full length of the variable in ``ds``,
+      then, again, use the full length from the dataset.
+    - If a dimension chunk is specified as -1, again, use the full length from the dataset.
```

Comment:
I may be missing something here, but is `-1` handled in the current implementation?

Reply:
It is in [xarrays `chunk`](https://xarray.pydata.org/en/v0.14.0/generated/xarray.DataArray.chunk.html), it is however not super well documented it seems.